test/powershell/engine/Basic/NativeCommandErrorHandling.Tests.ps1

```diff
@@ -0,0 +1,179 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Functional tests to verify that errors (non-terminating and terminating) appropriately
+# when $PSNativeCommandUseErrorActionPreference is $true
+
+Describe 'Native command error handling tests' -Tags 'CI' {
+    BeforeAll {
+        $originalDefaultParameterValues = $PSDefaultParameterValues.Clone()
+        if (-not [ExperimentalFeature]::IsEnabled('PSNativeCommandErrorActionPreference'))
+        {
+            $PSDefaultParameterValues['It:Skip'] = $true
+            return
+        }
+
+        $exeName = $IsWindows ? 'testexe.exe' : 'testexe'
+
+        $errorActionPrefTestCases = @(
+            @{ ErrorActionPref = 'Stop' }
+            @{ ErrorActionPref = 'Continue' }
+            @{ ErrorActionPref = 'SilentlyContinue' }
+            @{ ErrorActionPref = 'Ignore' }
+        )
+    }
+
+    AfterAll {
+        $global:PSDefaultParameterValues = $originalDefaultParameterValues
+    }
+
+    BeforeEach {
+        $Error.Clear()
+    }
+
+    Context 'PSNativeCommandUseErrorActionPreference is $true' {
+        BeforeEach {
+            $PSNativeCommandUseErrorActionPreference = $true
+        }
+
+        It 'Non-zero exit code throws teminating error for $ErrorActionPreference = ''Stop''' {
+            $ErrorActionPreference = 'Stop'
+
+            { testexe -returncode 1 } | Should -Throw -ErrorId 'ProgramFailedToComplete'
```

Comment:
The test shows the name is not good - the utility really _executed_ and _completed_ but return non-zero return code.
I think we need better name like NativeCommandReturnNonZeroReturnCode.

Reply:
Yeah that's a fair point. @theJasonHelmick, @rkeithhill we should consider renaming this error ID